wgsl/index.bs

```diff
@@ -4180,6 +4180,242 @@ TODO: deduplicate these tables
+textureSample(texture : texture_depth_2d_array,           sampler : sampler, coords : vec2<f32>, array_index : u32                    ) -> f32
+textureSample(texture : texture_depth_2d_array,           sampler : sampler, coords : vec2<f32>, array_index : u32, offset : vec2<i32>) -> f32
+textureSample(texture : texture_depth_cube,               sampler : sampler, coords : vec3<f32>                                       ) -> f32
+textureSample(texture : texture_depth_cube_array,         sampler : sampler, coords : vec3<f32>, array_index : u32                    ) -> f32
+```
+
+**Parameters:**
+
+<table class='data'>
+  <tr><td>`texture`<td>
+  The [sampled](#sampled-texture-type) or [depth](#texture-depth) texture to
+  sample.
+  <tr><td>`sampler`<td>
+  The [sampler type](#sampler-type).
+  <tr><td>`coords`<td>
+  The texture coordinates used for sampling.
+  <tr><td>`array_index`<td>
+  The 0-based texture array index to sample.
+  <tr><td>`offset`<td>
+  The optional texel offset applied to the unnormalized texture coordinate
+  before sampling the texture. This offset is applied before applying any
+  texture wrapping modes. `offset` values may range from `[-8..7]` in all
+  dimensions. Values outside of this range will be clamped to the nearest limit.
+</table>
+
+**Returns:**
+
+The sampled value.
+
+
+### `textureSampleBias` ### {#texturesamplebias}
+
+Samples a texture with a bias to the mip level.
+
+```rust
+textureSampleBias(texture : texture_sampled_2d<type>,         sampler : sampler, coords : vec2<f32>,                    bias : f32                    ) -> <type>
+textureSampleBias(texture : texture_sampled_2d<type>,         sampler : sampler, coords : vec2<f32>,                    bias : f32, offset : vec2<i32>) -> <type>
+textureSampleBias(texture : texture_sampled_2d_array<type>,   sampler : sampler, coords : vec2<f32>, array_index : u32, bias : f32                    ) -> <type>
+textureSampleBias(texture : texture_sampled_2d_array<type>,   sampler : sampler, coords : vec2<f32>, array_index : u32, bias : f32, offset : vec2<i32>) -> <type>
+textureSampleBias(texture : texture_sampled_3d<type>,         sampler : sampler, coords : vec3<f32>,                    bias : f32                    ) -> <type>
+textureSampleBias(texture : texture_sampled_3d<type>,         sampler : sampler, coords : vec3<f32>,                    bias : f32, offset : vec3<i32>) -> <type>
+textureSampleBias(texture : texture_sampled_cube<type>,       sampler : sampler, coords : vec3<f32>,                    bias : f32                    ) -> <type>
+textureSampleBias(texture : texture_sampled_cube_array<type>, sampler : sampler, coords : vec3<f32>, array_index : u32, bias : f32                    ) -> <type>
+textureSampleBias(texture : texture_depth_2d,                 sampler : sampler, coords : vec2<f32>,                    bias : f32                    ) -> f32
```

Comment:
I do wonder if we need these. Main purpose of the depth textures is comparison sampling. And for it, we don't even get to select the mipmap level (due to Metal limitations), so it's always forced 0. This means there is no incentive for the users to even provide texture views spanning more than one mipmap level.

Reply:
Removed